lib/web_ui/test/ui/line_metrics_test.dart

```diff
@@ -116,4 +116,23 @@ Future<void> testMain() async {
         expect(metrics, hasLength(1));
     }
   }, skip: isHtml); // The rounding hack doesn't apply to the html renderer
+
+  test('uses flutter test fonts when debugEmulateFlutterTesterEnvironment is enabled', () {
+    final ui.ParagraphBuilder builder = ui.ParagraphBuilder(ui.ParagraphStyle());
+    builder.pushStyle(ui.TextStyle(
+      fontSize: 10.0,
+      fontFamily: 'SomeOtherFontFamily',
+    ));
+    builder.addText('XXXX');
+    final ui.Paragraph paragraph = builder.build();
+    paragraph.layout(const ui.ParagraphConstraints(width: 400));
+
+    expect(paragraph.numberOfLines, 1);
+
+    final ui.LineMetrics? metrics = paragraph.getLineMetricsAt(0);
+    expect(metrics, isNotNull);
+
+    // Ahem's 'X' character is a square, so it's the font size (10.0) * 4 characters.
```

Comment:
Isn't the fallback font now `FlutterTest`? (`_testFonts.first`), or am I misunderstanding the testing fallback?

(If it's not Ahem, then I guess this comment is a little lie.)

PS: is there any way of adding an `expect`ation of what font was eventually used?

Reply:
I think `FlutterTest` is actually `Ahem`. It's the same font with a different name. Why do we have a `FlutterTest` font that is actually `Ahem`? I don't really know the historical reason. I'm just trying to match the behavior of CanvasKit

Reply:
Looked up some more context, which can be found in this PR: https://github.com/flutter/engine/pull/39809

The `FlutterTest` is *almost* `Ahem`, with a few modifications. I'm changing the comment to be more accurate.

Reply:
Oh, and in answer to your second question, no. The font is substituted under the hood in a way that isn't directly visible to a client via `dart:ui`. So we're doing measurements instead.